autogain/autogain.py

```diff
@@ -285,35 +306,45 @@ def process(self, fband, taper):
             unskipped = 0
             for i_s, s in enumerate(self.stations):
                 dist = distance_accurate50m(event, s)
-                arrival = self.phaser.t(self.phase_selection, (event.depth, dist))
-                if arrival==None:
-                    skipped +=1
-                    logger.debug('skipping event %s at stations %s. Reason no phase arrival'
-                                % (event, s))
+                arrival = self.phaser.t(self.phase_selection,
+                                        (event.depth, dist))
+                if arrival is None:
+                    skipped += 1
+                    logger.debug('skipping event %s at stations %s. ' +
+                                 'Reason no phase arrival'
+                                 % (event, s))
                     continue
                 else:
-                    unskipped +=1
-                selector = lambda tr: util.match_nslc('%s.*%s'%(s.nsl_string(),
-                                                                self.component),
+                    unskipped += 1
+                selector = lambda tr: util.match_nslc('%s.*%s' %
+                                                      (s.nsl_string(),
+                                                       self.component),
                                                       tr.nslc_id)
                 window_min, window_max = self.window.t()
-                tr = self.data_pile.chopper(tmin=event.time+arrival - window_min,
-                                            tmax=event.time+arrival + window_max,
+                tr = self.data_pile.chopper(tmin=event.time+arrival -
+                                            window_min,
+                                            tmax=event.time+arrival +
+                                            window_max,
                                             trace_selector=selector)
-                _tr = tr.next()
+                try:
+                    _tr = next(tr)    # does not work without try and except?!
```

Comment:
Hmm.... was I doing the same thing as in line 342 here? Probably I accidentally skipped every second trace :-(

Reply:
Ah, no that was just to make sure that each chunk of data containes just one trace. Puh, ugly way of doing this though...

Reply:
Nothing is skipped, but I guess it really is not a beautiful way to check for having only one trace. It works fine, though, but without the added try and except I had problems when there was no next trace anymore.

Reply:
Ah, sorry, thought that was my creation :-)